# The first part of the assignment

In [112]:
import pandas as pd
from pandas import Series,DataFrame
import numpy as np

In [113]:
filename ="https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M"

In [114]:
dfs = pd.read_html(filename)
df=dfs[0]
df.columns = ['Postal code', 'Borough', 'Neighborhood']
df.head()

Postal code           Borough               Neighborhood
0         M1A      Not assigned                        NaN
1         M2A      Not assigned                        NaN
2         M3A        North York                  Parkwoods
3         M4A        North York           Victoria Village
4         M5A  Downtown Toronto  Regent Park, Harbourfront

In [115]:
# Ignore cells with a borough that is Not assigned
df=df[df['Borough']!= 'Not assigned']
df.head(15)

Postal code           Borough  \
2          M3A        North York   
3          M4A        North York   
4          M5A  Downtown Toronto   
5          M6A        North York   
6          M7A  Downtown Toronto   
8          M9A         Etobicoke   
9          M1B       Scarborough   
11         M3B        North York   
12         M4B         East York   
13         M5B  Downtown Toronto   
14         M6B        North York   
17         M9B         Etobicoke   
18         M1C       Scarborough   
20         M3C        North York   
21         M4C         East York   

                                         Neighborhood  
2                                           Parkwoods  
3                                    Victoria Village  
4                           Regent Park, Harbourfront  
5                    Lawrence Manor, Lawrence Heights  
6         Queen's Park, Ontario Provincial Government  
8                                    Islington Avenue  
9                                      Malvern, Rouge  
11                                          Don Mills  
12                    Parkview Hill, Woodbine Gardens  
13                           Garden District, Ryerson  
14                                          Glencairn  
17  West Deane Park, Princess Gardens, Martin Grov...  
18             Rouge Hill, Port Union, Highland Creek  
20                                          Don Mills  
21                                   Woodbine Heights

In [116]:
# combine neighborhoods that exist in one postal code area

df.loc[df['Neighborhood'] =='Not assigned' , 'Neighborhood'] = df['Borough']

D:\Anaconda3\lib\site-packages\pandas\core\indexing.py:670: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_with_indexer(indexer, value)
D:\Anaconda3\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until


In [41]:
# use the neighborhood to replace a borough with Not assigned value
df.Neighborhood.replace("Not assigned", df.Borough, inplace=True)
df.head(15)

D:\Anaconda3\lib\site-packages\pandas\core\generic.py:6746: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._update_inplace(new_data)


Postal code           Borough  \
2          M3A        North York   
3          M4A        North York   
4          M5A  Downtown Toronto   
5          M6A        North York   
6          M7A  Downtown Toronto   
8          M9A         Etobicoke   
9          M1B       Scarborough   
11         M3B        North York   
12         M4B         East York   
13         M5B  Downtown Toronto   
14         M6B        North York   
17         M9B         Etobicoke   
18         M1C       Scarborough   
20         M3C        North York   
21         M4C         East York   

                                         Neighborhood  
2                                           Parkwoods  
3                                    Victoria Village  
4                           Regent Park, Harbourfront  
5                    Lawrence Manor, Lawrence Heights  
6         Queen's Park, Ontario Provincial Government  
8                                    Islington Avenue  
9                                      Malvern, Rouge  
11                                          Don Mills  
12                    Parkview Hill, Woodbine Gardens  
13                           Garden District, Ryerson  
14                                          Glencairn  
17  West Deane Park, Princess Gardens, Martin Grov...  
18             Rouge Hill, Port Union, Highland Creek  
20                                          Don Mills  
21                                   Woodbine Heights

In [42]:
for index, row in df.iterrows():
    if row["Neighborhood"] == "Not assigned":
        row["Neighborhood"] = row["Borough"]

In [43]:
df.head(15)

Postal code           Borough  \
2          M3A        North York   
3          M4A        North York   
4          M5A  Downtown Toronto   
5          M6A        North York   
6          M7A  Downtown Toronto   
8          M9A         Etobicoke   
9          M1B       Scarborough   
11         M3B        North York   
12         M4B         East York   
13         M5B  Downtown Toronto   
14         M6B        North York   
17         M9B         Etobicoke   
18         M1C       Scarborough   
20         M3C        North York   
21         M4C         East York   

                                         Neighborhood  
2                                           Parkwoods  
3                                    Victoria Village  
4                           Regent Park, Harbourfront  
5                    Lawrence Manor, Lawrence Heights  
6         Queen's Park, Ontario Provincial Government  
8                                    Islington Avenue  
9                                      Malvern, Rouge  
11                                          Don Mills  
12                    Parkview Hill, Woodbine Gardens  
13                           Garden District, Ryerson  
14                                          Glencairn  
17  West Deane Park, Princess Gardens, Martin Grov...  
18             Rouge Hill, Port Union, Highland Creek  
20                                          Don Mills  
21                                   Woodbine Heights

In [44]:
# print the number of rows of your dataframe.
df.shape

(103, 3)

# The second part

In [45]:
# get the geographical coordinates 
loc_info = pd.read_csv('https://cocl.us/Geospatial_data')
loc_info.columns = ['Postal code','Latitude','Longitude']
loc_info.head()

Postal code   Latitude  Longitude
0         M1B  43.806686 -79.194353
1         M1C  43.784535 -79.160497
2         M1E  43.763573 -79.188711
3         M1G  43.770992 -79.216917
4         M1H  43.773136 -79.239476

In [46]:
# merge data to get the desired dataframe
df_all = pd.merge(df, loc_info, on = 'Postal code')
df_all.head(15)

Postal code           Borough  \
0          M3A        North York   
1          M4A        North York   
2          M5A  Downtown Toronto   
3          M6A        North York   
4          M7A  Downtown Toronto   
5          M9A         Etobicoke   
6          M1B       Scarborough   
7          M3B        North York   
8          M4B         East York   
9          M5B  Downtown Toronto   
10         M6B        North York   
11         M9B         Etobicoke   
12         M1C       Scarborough   
13         M3C        North York   
14         M4C         East York   

                                         Neighborhood   Latitude  Longitude  
0                                           Parkwoods  43.753259 -79.329656  
1                                    Victoria Village  43.725882 -79.315572  
2                           Regent Park, Harbourfront  43.654260 -79.360636  
3                    Lawrence Manor, Lawrence Heights  43.718518 -79.464763  
4         Queen's Park, Ontario Provincial Government  43.662301 -79.389494  
5                                    Islington Avenue  43.667856 -79.532242  
6                                      Malvern, Rouge  43.806686 -79.194353  
7                                           Don Mills  43.745906 -79.352188  
8                     Parkview Hill, Woodbine Gardens  43.706397 -79.309937  
9                            Garden District, Ryerson  43.657162 -79.378937  
10                                          Glencairn  43.709577 -79.445073  
11  West Deane Park, Princess Gardens, Martin Grov...  43.650943 -79.554724  
12             Rouge Hill, Port Union, Highland Creek  43.784535 -79.160497  
13                                          Don Mills  43.725900 -79.340923  
14                                   Woodbine Heights  43.695344 -79.318389

# The third part

Choose boroughs that contain the word Toronto

In [47]:
df_Toronto = df_all[df_all['Borough'].str.contains('Toronto',regex=False)]
df_Toronto.head()

Postal code           Borough                                 Neighborhood  \
2          M5A  Downtown Toronto                    Regent Park, Harbourfront   
4          M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government   
9          M5B  Downtown Toronto                     Garden District, Ryerson   
15         M5C  Downtown Toronto                               St. James Town   
19         M4E      East Toronto                                  The Beaches   

     Latitude  Longitude  
2   43.654260 -79.360636  
4   43.662301 -79.389494  
9   43.657162 -79.378937  
15  43.651494 -79.375418  
19  43.676357 -79.293031

Import necessary Libraries

In [48]:
import requests # library to handle requests
import random # library for random number generation
!conda install -c conda-forge geopy --yes
from geopy.geocoders import Nominatim # module to convert an address into latitude and longitude values
# libraries for displaying images
from IPython.display import Image
from IPython.core.display import HTML
# tranforming json file into a pandas dataframe library
from pandas.io.json import json_normalize
!conda install -c conda-forge folium=0.5.0 --yes
import folium # plotting library
print('Folium installed')
print('Libraries imported.')

Solving environment: ...working... done

# All requested packages already installed.

Solving environment: ...working... done

# All requested packages already installed.

Folium installed
Libraries imported.


Define Foursquare Credentials and Version

In [49]:
CLIENT_ID = '' # I deleted these two lines after I finish the project for privacy
CLIENT_SECRET = '' 
VERSION = '20180604'
LIMIT = 30

Suppose we are shopping in a Mall and want to find a Chinese restaurant nearby to have lunch. The following work is used for locating all the Chinese restuarants around and helps us to know the ratings

In [50]:
# Choose a random mall in downtown Toronto: Dragon City Mall

address = '280 Spadina Ave, Toronto, ON'
geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)

43.6526635 -79.3982828


Search for a specific venue category：Chinese restaurant

In [51]:
search_query = 'Chinese restaurant'
radius = 300
print(search_query + ' .... OK!')

Chinese restaurant .... OK!


In [109]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)

In [53]:
results = requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5ec72f6398205d001b9f1473'},
 'response': {'venues': [{'id': '4b072e9df964a52009f922e3',
    'name': 'Sky Dragon Chinese Restaurant 龍翔酒樓',
    'location': {'address': '280 Spadina Ave.',
     'crossStreet': 'at Dundas St. W.',
     'lat': 43.65278331265585,
     'lng': -79.39817372355205,
     'labeledLatLngs': [{'label': 'display',
       'lat': 43.65278331265585,
       'lng': -79.39817372355205}],
     'distance': 15,
     'cc': 'CA',
     'city': 'Toronto',
     'state': 'ON',
     'country': 'Canada',
     'formattedAddress': ['280 Spadina Ave. (at Dundas St. W.)',
      'Toronto ON',
      'Canada']},
    'categories': [{'id': '4bf58dd8d48988d1f5931735',
      'name': 'Dim Sum Restaurant',
      'pluralName': 'Dim Sum Restaurants',
      'shortName': 'Dim Sum',
      'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/dimsum_',
       'suffix': '.png'},
      'primary': True}],
    'referralId': 'v-1590112427',
    'hasPerk': False}

In [54]:
# assign relevant part of JSON to venues
venues = results['response']['venues']


In [57]:
# tranform venues into a dataframe
df2 = pd.json_normalize(venues)
df2.head(40)

id                                             name  \
0   4b072e9df964a52009f922e3               Sky Dragon Chinese Restaurant 龍翔酒樓   
1   4d06c1ccc2e53704428cc567               North-East Chinese Restaurant 華北美食   
2   52117a4c11d2ac3f5ac117a3    Traditional Northern Style Chinese Restaurant   
3   57b8e03a498e48d4813ab5af                       Harmony Chinese Restaurant   
4   4b622bf1f964a5200b3a2ae3                         Chinese Traditional Buns   
5   4ae7710bf964a52057ab21e3                               Canteen Restaurant   
6   4b266f05f964a520657b24e3                   Goldstone Noodle Restaurant 金石   
7   5c33c7cd8194fc002c66f72e                    Green Tea Restaurant Downtown   
8   5aefb64a491be7002c4bd30b  Ontario College of Traditional Chinese Medicine   
9   57068b11498e49429060c0de                           Savoury Chinese Crêpes   
10  4ae74cfaf964a5202eaa21e3                         King's Noodle House 富豪麵家   
11  4fd3ea3ee4b0191b9c2b1aff                         Rosewood Chinese Cuisine   
12  4b772c00f964a52097842ee3                       Sang Ho Seafood Restaurant   
13  51b604a0498e2384782ee1ef        Yoo J Express Restaurant (Purple Place! )   
14  4ae29812f964a520288f21e3                          Swatow Restaurant 汕頭小食家   
15  4c3796eb18e72d7f63d216f5                            Chinese Gospel Church   
16  4f036189c5126f14bacbf103                                    Chinese Herbs   
17  534aaebf498eee83e7fd2e7c                   Chinese Freemasons Association   
18  4bce1ec0cc8cd13a9ac8c3cf                Kensington Cornerstone Restaurant   
19  4ba2d00cf964a5203e1b38e3                         Ka Chi Korean Restaurant   
20  4b074bb1f964a52077fb22e3                         New Sky Restaurant 小沙田食家   
21  4ad4c060f964a5207ff720e3                           Rol San Restaurant 龍笙棧   
22  4af89f6ff964a520c30e22e3                                   Chinese Bakery   
23  59eccc99dff8150ccdd70355                        Kim Vietnamese Restaurant   
24  4f0f907ce4b067f66dbc8a05                              Rong Hua Restaurant   
25  5bca8bc3ccad6b002c8dd841                       BiWon Korean Restaurant 秘園   
26  4ba573a5f964a5207c0839e3                       Rol Jui Seafood Restaurant   
27  4c2389fe13c00f47563189de     Goodman Chinese Natural Herbs And Food Store   
28  4af1c3a4f964a520f2e221e3                 Bun Saigon Vietnamese Restaurant   
29  4b119787f964a520188023e3                                   Dumpling House   

                                           categories    referralId  hasPerk  \
0   [{'id': '4bf58dd8d48988d1f5931735', 'name': 'D...  v-1590112427    False   
1   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
2   [{'id': '4bf58dd8d48988d142941735', 'name': 'A...  v-1590112427    False   
3   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
4   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
5   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
6   [{'id': '4bf58dd8d48988d1d1941735', 'name': 'N...  v-1590112427    False   
7   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
8   [{'id': '4bf58dd8d48988d1ad941735', 'name': 'T...  v-1590112427    False   
9   [{'id': '56aa371be4b08b9a8d57350b', 'name': 'F...  v-1590112427    False   
10  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
11  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
12  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
13  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
14  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
15  [{'id': '4bf58dd8d48988d132941735', 'name': 'C...  v-1590112427    False   
16  [{'id': '4bf58dd8d48988d1ff941735', 'name': 'M...  v-1590112427    False   
17  [{'id': '52e81612bcbc57f1066b7a35', 'name': 'C...  v-1590112427    Fal

In [65]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns = ['name', 'categories'] + [col for col in df2.columns if col.startswith('location.')] + ['id']
df2_filtered = df2.loc[:, filtered_columns]

In [66]:
# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']


In [67]:
# filter the category for each row
df2_filtered['categories'] = df2_filtered.apply(get_category_type,axis=1)
# clean column names by keeping only last term
df2_filtered.columns = [column.split('.')[-1] for column in df2_filtered.columns]
df2_filtered

name             categories  \
0                Sky Dragon Chinese Restaurant 龍翔酒樓     Dim Sum Restaurant   
1                North-East Chinese Restaurant 華北美食     Chinese Restaurant   
2     Traditional Northern Style Chinese Restaurant       Asian Restaurant   
3                        Harmony Chinese Restaurant     Chinese Restaurant   
4                          Chinese Traditional Buns     Chinese Restaurant   
5                                Canteen Restaurant     Chinese Restaurant   
6                    Goldstone Noodle Restaurant 金石           Noodle House   
7                     Green Tea Restaurant Downtown     Chinese Restaurant   
8   Ontario College of Traditional Chinese Medicine           Trade School   
9                            Savoury Chinese Crêpes             Food Stand   
10                         King's Noodle House 富豪麵家     Chinese Restaurant   
11                         Rosewood Chinese Cuisine     Chinese Restaurant   
12                       Sang Ho Seafood Restaurant     Chinese Restaurant   
13        Yoo J Express Restaurant (Purple Place! )     Chinese Restaurant   
14                          Swatow Restaurant 汕頭小食家     Chinese Restaurant   
15                            Chinese Gospel Church                 Church   
16                                    Chinese Herbs     Miscellaneous Shop   
17                   Chinese Freemasons Association             Club House   
18                Kensington Cornerstone Restaurant         Breakfast Spot   
19                         Ka Chi Korean Restaurant      Korean Restaurant   
20                         New Sky Restaurant 小沙田食家     Chinese Restaurant   
21                           Rol San Restaurant 龍笙棧     Dim Sum Restaurant   
22                                   Chinese Bakery                 Bakery   
23                        Kim Vietnamese Restaurant  Vietnamese Restaurant   
24                              Rong Hua Restaurant     Chinese Restaurant   
25                       BiWon Korean Restaurant 秘園      Korean Restaurant   
26                       Rol Jui Seafood Restaurant     Chinese Restaurant   
27     Goodman Chinese Natural Herbs And Food Store                   None   
28                 Bun Saigon Vietnamese Restaurant  Vietnamese Restaurant   
29                                   Dumpling House    Dumpling Restaurant   

                             address                     crossStreet  \
0                   280 Spadina Ave.                at Dundas St. W.   
1                     476 Dundas St.                           Huron   
2                                NaN                             NaN   
3             478 Dundas Street West                             NaN   
4                   536 Dundas St. W                  at Spadina Ave   
5   280 Spadina Ave, Unit L, 2nd Fl.                  at Dundas St W   
6                    266 Spadina Ave                  at Willison Sq   
7    261 Spadina Avenue. Upper level                             NaN   
8                   283 Spadina Ave.                             NaN   
9                    492 Dundas St W                     Spadina Ave   
10                   296 Spadina Ave                  at Dundas St W   
11                 463 Dundas St. W.                    at Larch St.   
12                   536 Dundas St W                  Dundas/Spadina   
13                   280 Spadina Ave                      Dundas St.   
14                  309 Spadina Ave.  btwn Dundas St. W & D'Arcy St.   
15            450 Dundas Street West                             NaN   
16                      Saint Andrew                             NaN   
17                   436 Dundas St W                             NaN   
18                2A Kensington Ave.                    Dundas St. W   
19                   8 St Andrew St.                 at Spadina Ave.   
20                  353 Spadina Ave.                             NaN   
21                  323 Spadina Ave.        

In [91]:
# keep only the Chinese Restaurants in catagories
df3=df2[df2_filtered['categories']=='Chinese Restaurant']
df3

id                                       name  \
1   4d06c1ccc2e53704428cc567         North-East Chinese Restaurant 華北美食   
3   57b8e03a498e48d4813ab5af                 Harmony Chinese Restaurant   
4   4b622bf1f964a5200b3a2ae3                   Chinese Traditional Buns   
5   4ae7710bf964a52057ab21e3                         Canteen Restaurant   
7   5c33c7cd8194fc002c66f72e              Green Tea Restaurant Downtown   
10  4ae74cfaf964a5202eaa21e3                   King's Noodle House 富豪麵家   
11  4fd3ea3ee4b0191b9c2b1aff                   Rosewood Chinese Cuisine   
12  4b772c00f964a52097842ee3                 Sang Ho Seafood Restaurant   
13  51b604a0498e2384782ee1ef  Yoo J Express Restaurant (Purple Place! )   
14  4ae29812f964a520288f21e3                    Swatow Restaurant 汕頭小食家   
20  4b074bb1f964a52077fb22e3                   New Sky Restaurant 小沙田食家   
24  4f0f907ce4b067f66dbc8a05                        Rong Hua Restaurant   
26  4ba573a5f964a5207c0839e3                 Rol Jui Seafood Restaurant   

                                           categories    referralId  hasPerk  \
1   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
3   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
4   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
5   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
7   [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
10  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
11  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
12  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
13  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
14  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
20  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
24  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   
26  [{'id': '4bf58dd8d48988d145941735', 'name': 'C...  v-1590112427    False   

                    location.address            location.crossStreet  \
1                     476 Dundas St.                           Huron   
3             478 Dundas Street West                             NaN   
4                   536 Dundas St. W                  at Spadina Ave   
5   280 Spadina Ave, Unit L, 2nd Fl.                  at Dundas St W   
7    261 Spadina Avenue. Upper level                             NaN   
10                   296 Spadina Ave                  at Dundas St W   
11                 463 Dundas St. W.                    at Larch St.   
12                   536 Dundas St W                  Dundas/Spadina   
13                   280 Spadina Ave                      Dundas St.   
14                  309 Spadina Ave.  btwn Dundas St. W & D'Arcy St.   
20                  353 Spadina Ave.                             NaN   
24                   478 Dundas St W                             NaN   
26                     472 Dundas St                         Spadina   

    location.lat  location.lng  \
1      43.653185    -79.396677   
3      43.653427    -79.397030   
4      43.652714    -79.399006   
5      43.652821    -79.398275   
7      43.652488    -79.397501   
10     43.653089    -79.398214   
11     43.653171    -79.396710   
12     43.652747    -79.399021   
13     43.652564    -79.398903   
14     43.653866    -79.398334   
20     43.655337    -79.398897   
24     43.653144    -79.397133   
26     43.653153    -79.396877   

                              location.labeledLatLngs  location.distance  \
1   [{'label': 'display', 'lat': 43.6531846832903,...                141   
3   [{'label': 'display', 'lat': 43.653427, 'lng':...                131   
4   [{'label': 'display', 'lat': 43.65271411616852...                 58   
5   [{'label': 'display', 'lat': 43.6528214686215,...           

In [79]:
venues_map = folium.Map(location=[latitude, longitude], zoom_start=13) #generate map 
# add a red circle marker to represent the Dragon City Mall
folium.features.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='Dragon City Mall',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.3
).add_to(venues_map)

In [93]:
for lat, lng, label in zip(df2_filtered.lat, df2_filtered.lng, df2_filtered.categories):
    folium.features.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.3
    ).add_to(venues_map)
    
# display map
venues_map

Explore the rating of a Given Venue

In [107]:
# the function to get the ratings of the restaurants by using their IDs
def ratings(venue_id):
    url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID,CLIENT_SECRET, VERSION)
    result = requests.get(url).json()
    result['response']['venue']
    try:
        print(result['response']['venue']['rating'])
    except:
        print('This venue has not been rated yet.')

In [108]:
ratings ('4d06c1ccc2e53704428cc567') # ratings of the first Chinese Restaurant North-East Chinese Restaurant

6.3


6.3 is a relatively good rating. We can continue to explore the ratings of other restaurants' ratings if needed